### PR TITLE
rmw_take_sequence() implementation

### PR DESCRIPTION
**Pull request by pablogs9**
This PR implements function `rmw_take_sequence` in `rmw-microxrcedds`

### CODE REVIEW

File: rmw_microxrcedds_c/src/rmw_take.c [hunk 1]
```diff
@@ -88,15 +88,45 @@ rmw_take_sequence(
...
+      break;
+    }
+
+    if (taken_flag) {
+      (*taken)++;
+    }
+  }
+
+  message_sequence->size = *taken;
+  message_info_sequence->size = *taken;
```
Comment (julibert) [inline rmw_microxrcedds_c/src/rmw_take.c:127]: Where `message_info_sequence` is set? Is it correct to set the size equal to `taken`?

File: rmw_microxrcedds_c/src/rmw_take.c [hunk 2]
```diff
@@ -88,15 +88,45 @@ rmw_take_sequence(
...
+      &taken_flag,
+      allocation
+    );
+
+    if (ret != RMW_RET_OK) {
+      break;
+    }
+
+    if (taken_flag) {
+      (*taken)++;
```
Comment (julibert) [inline rmw_microxrcedds_c/src/rmw_take.c:122]: Should we `break` when `taken_flag` is false?

File: rmw_microxrcedds_c/src/rmw_take.c (hunk 1 above)
Comment (pablogs9) [reply rmw_microxrcedds_c/src/rmw_take.c:127]: `message_info_sequence` cames initializated from higher layer: https://github.com/ros2/rcl/blob/84cac5937b303e0dca3c9c1b8ffc8b73a2bd0197/rcl/src/rcl/subscription.c#L317

I guess that when you call `rmw_take_sequence` you expect to obtain a sequence of message with their appropriate message info, so the quantity of messages is equal to the quantity of message infos (one per each).

Check this:
https://github.com/ros2/rmw_fastrtps/blob/d1702b1ce0279fa0ce12acb382e14a9a611e4321/rmw_fastrtps_shared_cpp/src/rmw_take.cpp#L137

File: rmw_microxrcedds_c/src/rmw_take.c (hunk 2 above)
Comment (pablogs9) [reply rmw_microxrcedds_c/src/rmw_take.c:122]: We are `break`ing when `ret` is not `RMW_RET_OK`. In this case, the return code can be `RMW_RET_OK` and the `taken_flag` can be `false` if there is a deserialization error. Should we stop a sequence reading if one of the messages has failed in deserialization?

Comment (julibert) [reply rmw_microxrcedds_c/src/rmw_take.c:122]: I think so, in another case there would be an error and the use is not aware of that.

Comment (pablogs9) [reply rmw_microxrcedds_c/src/rmw_take.c]: Fixed

File: rmw_microxrcedds_c/src/rmw_take.c (hunk 1 above)
Comment (julibert) [reply rmw_microxrcedds_c/src/rmw_take.c:125]: But in the `fastrtps` case, they take the info also (check [here](https://github.com/ros2/rmw_fastrtps/blob/d1702b1ce0279fa0ce12acb382e14a9a611e4321/rmw_fastrtps_shared_cpp/src/rmw_take.cpp#L125)).

Comment (pablogs9) [reply rmw_microxrcedds_c/src/rmw_take.c:126]: Fixed

**Pull request merged**